lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -400,13 +375,44 @@ bool SPIRVRegularizeLLVMBase::regularize() {
         // a scalar/vector integers, so we have to extend such arguments
         if (II.isLogicalShift() &&
             II.getOperand(0)->getType()->isIntOrIntVectorTy(1)) {
-          auto *NewInst = extendBitInstBoolArg(&II);
-          for (auto *U : II.users()) {
-            if (cast<Instruction>(U)->getOpcode() == Instruction::ZExt) {
-              U->dropAllReferences();
-              U->replaceAllUsesWith(NewInst);
-              ToErase.push_back(cast<Instruction>(U));
+          IRBuilder<> Builder(&II);
+          Type *Op0Ty = II.getOperand(0)->getType();
```

Comment:
Nit: Op0Ty can be moved outside 'if' block and can be used inside the if conditional.

Reply:
I consider this whole branch as cold - even freezing so these instructions won't be executed often. 
To the line: I'm not sure how much overhead it introduces (maybe nothing). At first glance you have at least one touch of memory on heap which is unnecessary if the branch isn't taken.